File: COW/games/test/ais/pyAi/ai.py

class Ai(object):
	#Initialises the AI
	def init(self):
		#Do nothing
		nothing = 0
	
	#Executes the AI
	def execute(self, phase):
		#Phase: Test bool
		if phase == PHASE_TEST_BOOL:
			print "Test: F xor F =", xor(False, False)
			print "Test: F xor T =", xor(False, True)
			print "Test: T xor F =", xor(True, False)
			print "Test: T xor T =", xor(True, True)
			print "Test: 0 xor 1 =", xor(0, 1)
		
		#Phase : Test int
		elif phase == PHASE_TEST_INT:
			print "Test: 42++ =", plusplus(42)
			print "Test: -42++ =", plusplus(-42)
		
		#Phase : Test double
		elif phase == PHASE_TEST_DOUBLE:
			print "Test: 42/10 =", divide10(42)
			print "Test: -567/10 =", divide10(-567)
		
		#Phase : Test string
		elif phase == PHASE_TEST_STRING:
			print "Test: Hello.World =", concat("Hello", "World")
		
		#Phase : Test array
		elif phase == PHASE_TEST_ARRAY:
			print "Test: sum([3, 4, 5, 6, 7, 8, 9]) =", sum([3, 4, 5, 6, 7, 8, 9])
			values = count(10)
			print "Test: count(10) =", values
			print "Test: count(10)[7:] =", values[7:]
		
		#Phase : Test array
		elif phase == PHASE_TEST_VIEW:
			print "Test: printText('plop')"
			printText("plop")
	
	#Stops the AI
	def stop(self):
		#Do nothing
		nothing = 0

File: testAI/src/ai/Ai.java
/**
 * JavaTestAI - The AI implementation for the game "test".
 * 
 * @author Didjor
 */

package ai;

import test.Api;
import test.TestAi;

public class Ai implements TestAi {
	/**
	 * Initializes the AI before the game starts.
	 */
	public void init() {
		// Do nothing
	}
	
	/**
	 * Executes the test boolean phase.
	 */
	public void testBool() {
		System.out.println("F xor F = " + Api.xor(false, false));
		System.out.println("F xor T = " + Api.xor(false, true));
		System.out.println("T xor F = " + Api.xor(true, false));
		System.out.println("T xor T = " + Api.xor(true, true));
	}
	
	/**
	 * Executes the test integer phase.
	 */
	public void testInt() {
		System.out.println("42++ = " + Api.plusplus(42));
		System.out.println("-42++ = " + Api.plusplus(-42));
	}
	
	/**
	 * Executes the test double phase.
	 */
	public void testDouble() {
		System.out.println("42/10 = " + Api.divide10(42));
	}
	
	/**
	 * Executes the test string phase.
	 */
	public void testString() {
		System.out.println("Hello.World = " + Api.concat("Hello", "World"));
	}
	
	/**
	 * Executes the test array phase.
	 */
	public void testArray() {
		int[] values = { 3, 4, 5, 6, 7, 8, 9 };
		System.out.println("sum([3, 4, 5, 6, 7, 8, 9]) = " + Api.sum(values));
		
		int[] returnValues = Api.count(10);
		System.out.println("count(10)[7] = " + returnValues[7]);
	}
	
	/**
	 * Stops the AI.
	 */
	public void stop() {
		// Do nothing
	}
}
